extension/httpfs/httpfs.cpp

```diff
@@ -329,6 +329,14 @@ unique_ptr<ResponseWrapper> HTTPFileSystem::GetRangeRequest(FileHandle &handle,
 				    hfs.state->total_bytes_received += data_length;
 			    }
 			    if (buffer_out != nullptr) {
+				    if (data_length + out_offset > buffer_out_len) {
+					    // As of v0.8.2-dev4424 we might end up here when very big files are served from servers
+					    // that returns more data than requested via range header. This is an uncommon but legal
+					    // behaviour, so we have to improve logic elsewhere to properly handle this case.
+
+					    // To avoid corruption of memory, we bail out.
+					    return false;
+				    }
 				    memcpy(buffer_out + out_offset, data, data_length);
```

Comment:
What if instead of bailing out we just constrain the amount of data buffered to the buffer length, i.e. do something like:

```cpp
data_length = MinValue<idx_t>(data_length, buffer_out_len - data_length - out_offset);
```

That way we would just silently discard any data that is sent past the end of the buffer.